vnext/Microsoft.ReactNative/Modules/LinkingManagerModule.cpp

```diff
@@ -3,76 +3,100 @@
 
 #include "pch.h"
 
-#include <Utils/ValueUtils.h>
 #include <winrt/Windows.System.h>
+#include "JSValueWriter.h"
 #include "LinkingManagerModule.h"
-#include "Unicode.h"
-
-#include <cxxreact/Instance.h>
-#include <cxxreact/JsArgumentHelpers.h>
+#include "ReactHost/React.h"
 
 #if _MSC_VER <= 1913
 // VC 19 (2015-2017.6) cannot optimize co_await/cppwinrt usage
 #pragma optimize("", off)
 #endif
 
-using Callback = facebook::xplat::module::CxxModule::Callback;
-
 using namespace winrt;
 using namespace Windows::Foundation;
 using namespace Windows::System;
-using namespace ::Microsoft::Common::Unicode;
 
 namespace Microsoft::ReactNative {
 
 //
 // LinkingManagerModule helpers
 //
 
-static fire_and_forget openURLAsync(Uri uri, Callback success, Callback error) {
+static fire_and_forget openURLAsync(Uri uri, React::ReactPromise<React::JSValue> result) {
   if (co_await Launcher::LaunchUriAsync(uri)) {
-    success({true});
+    result.Resolve({true});
   } else {
-    error({folly::dynamic::object("code", 1)("message", "Unable to open URL: " + Utf16ToUtf8(uri.DisplayUri()))});
+    const auto message = L"Unable to open URL: " + uri.DisplayUri();
+    result.Reject(message.c_str());
```

Comment:
Not sure if it's a problem to change the `"code"` here. I believe this changes to `EUNSPECIFIED`

Reply:
Seems fine.